velox/common/fuzzer/ConstrainedGenerators.h

```diff
@@ -375,4 +388,52 @@ class JsonInputGenerator : public AbstractInputGenerator {
 
   folly::json::serialization_opts opts_;
 };
+
+/// Generates a JSON path for JSON of a given type.
+/// @param jsonType The type of data represented by the JSON.
+/// @param mapKeys Candidate key values of maps in the JSON. All maps in the
+/// JSON are assumed to share the same key type and candidate key values.
+/// @param maxContainerLength The maximum length of a container (array or map)
+/// in the JSON.
+/// @param makeRandomVariation If true, the generator will generate JSON path
+/// not supported by JsonExtractor.
+/// This generator generates the following JSON paths:
+///  - On root: $
+///  - On strings: .length()
+///  - On arrays: .length(), [index], [], [begin:end:step]
```

Comment:
- `[]` => is invalid, we can keep this in case the intention here was to deliberately have invalid path.
- `[begin:end:step]` => we currently do not support this, can we remove it?
- we can also add negative indices
- `$.1` => is also valid where `1` is the index
- we can also use the wildcard operator `*` for arrays

lets start with valid paths, then we can incrementally add invalid paths